saleor/graphql/invoice/mutations.py

```diff
@@ -274,4 +274,5 @@ def perform_mutation(cls, _root, info, **data):
         cls.clean_instance(info, instance)
         send_invoice.delay(instance.pk)
         events.invoice_sent_event(user=info.context.user, invoice=instance)
+        info.context.plugins.invoice_sent(instance, instance.order.user.email)
```

Comment:
Don't you think that this call and `events.invoice_sent_event` should be called inside `send_invoice.delay` ?
We don't know the time when celery worker will process this task. This could cause strange cases like:
 - we have events that invoice was send but we didn't send it because our celery worker is under the huge load
 - we have events that invoice was send but sending an email for some reason failed.

Reply:
Very good point! Will defienietely move that.